### ISSUE TITLE
JSON fetching/unmarshalling error in PiHole API response

### ISSUE THREAD

**Issue by anirudhgray** [glanceapp/glance#289]
The DNS widget returns the following error when trying to use it with PiHole: `json: cannot unmarshal array into Go struct field piholeStatsResponse.domains_over_time of type map[int64]int`.

### Config
```
- type: dns-stats
            service: pihole
            url: http://tailscale-url
            hour_format: 24h
            token: TOKEN
```

### Environment
- glance: v0.6.4
- pihole: v5.18.3

Additionally, the `/admin/api.php?` returns a valid json object when tested directly in the browser with the api key. I am accessing everything via a tailscale VPN (pihole is running on another machine). Looking at the error message, it seems to be expecting a map in the `domains_over_time` property of the response, but is getting an array instead, which I can verify via the direct response in the browser: `"domains_over_time": [],`.

**Comment by anirudhgray** [comment]
This seems to have been an edge case. I switched on "Query Logging" in PiHole settings and as soon as some data came in, I started getting an appropriate map for `domains_over_time`, instead of an empty array:

````
"domains_over_time": {
"1735292700": 0,
...
}
```

I then switched query logging off, and flushed all logs, and everything is still OK. Maybe some sort of cold start problem?

Closing the issue since it is fixed.
